js/components/portal-dashboard/question-area.tsx

```diff
@@ -17,23 +17,20 @@ export class QuestionArea extends React.PureComponent<IProps>{
     const { currentQuestion, hideQuestion, useMinHeight } = this.props;
     const teacherEditionButtonClasses = css.teacherEditionButton;
     const teacherEditionBadge = css.teacherEditionBadge;
-    const prompt = currentQuestion?.get("prompt");
     const type = currentQuestion?.get("type");
     const scored = currentQuestion?.get("scored");
     const typeText = type && type.replace(/_/gm, ' ');
+    const interactiveName = currentQuestion?.get("name");
     const questionType = QuestionTypes.find(qt => qt.type === type && qt.scored === scored);
     const QuestionIcon = questionType?.icon;
-    const mcChoices: Map<any,any> = currentQuestion?.get("choices") || [];
-    const drawingPrompt = currentQuestion?.get("drawingPrompt");
-    const showDivider = drawingPrompt && prompt;
 
     return (
       <div className={`${css.questionArea} ${hideQuestion ? css.hidden : ""}`}>
         <div className={css.questionTypeHeader}>
           <div className={css.leftTitle}>
             <QuestionIcon className={`${css.icon} ${css.questionTypeIcon}`} />
             <span className={css.questionTypeTitle} data-cy="question-title">
-              {typeText}
+              {type === "iframe_interactive"? (interactiveName? interactiveName : typeText) : typeText}
```

Comment:
I think this will be more clear if in line 23 you have:
`const interactiveName = type === "iframe_interactive" && currentQuestion?.get("name");`

And then here in the render you just use::
`{ interactiveName || typeText }`

This way, you're more fully defining `interactiveName` based on both the type and the question and the render portion becomes more clear and doesn't require two ternaries.